packages/cadl-ranch/src/scenarios-resolver.ts

```diff
@@ -27,13 +27,31 @@ interface CadlRanchScenarioFile {
 
 export async function findScenarioCadlFiles(scenariosPath: string): Promise<CadlRanchScenarioFile[]> {
   await ensureScenariosPathExists(scenariosPath);
-  const pattern = `${normalizePath(scenariosPath)}/**/main.tsp`;
+  const pattern = [`${normalizePath(scenariosPath)}/**/client.tsp`, `${normalizePath(scenariosPath)}/**/main.tsp`];
   logger.debug(`Looking for scenarios in ${pattern}`);
   const scenarios = await findFilesFromPattern(pattern);
-  logger.info(`Found ${scenarios.length} scenarios.`);
+  scenarios.sort();
+  const result: string[] = [];
+  let shouldSkip = false;
+  scenarios.forEach((scenario, index) => {
+    if (shouldSkip) {
+      shouldSkip = false;
+    } else {
+      if (index + 1 < scenarios.length) {
+        if (scenario.replace("client.tsp", "main.tsp") === scenarios[index + 1]) {
```

Comment:
Does this mean that if it finds a `main.tsp` after a `client.tsp` then it skips it?

Reply:
yes, because I have sorted it, client,tsp should appear first.

Reply:
Actually I was thinking skip the next main.tsp when I found client.tsp, that's why I don't use filter at first, because it can only filter the current item, right ? but then I realize we can skip main.tsp and compare with the previous one see if it's client.tsp, then skip main.tsp when found main.tsp. That way filter would work.

Reply:
I would maybe instead use a set and do a getDirName and join main.tsp instead as it feels more reliable than just checking the previous item in the sorted list

Reply:
Okay, I changed this, not sure if I fully understood. anyway, let me know if it's not.